crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -1185,11 +1185,6 @@ export.save_account_storage_data
     # verify hashed account storage slots match account storage commitment
     assert_eqw.err=ERR_ACCOUNT_STORAGE_COMMITMENT_MISMATCH
     # OS => []
-
-    # duplicate the initial storage slots in memory to enable a diff computation
-    # for the account delta
-    exec.memory::mem_copy_initial_storage_slots
```

Comment:
Was this a bug previously? My understanding is that the `save_account_storage_data` procedure is called every time we load a foreign account, and this would imply that `mem_copy_initial_storage_slots` would be called every time we load a foreign account as well - right?